diff --git a/dist/mgba.js b/dist/mgba.js
index ceede94cdd1365f3c472fd75aa6f48ec0bb661ea..73e9b3f251b2422aa458ecf463d36b51de77e832 100644
--- a/dist/mgba.js
+++ b/dist/mgba.js
@@ -1601,7 +1601,7 @@ var ASM_CONSTS = {
         // instead of just using new URL(import.meta.url) because bundler's only recognize
         // the first case in their bundling step. The latter ends up producing an invalid
         // URL to import from the server (e.g., for webpack the file:// path).
-        worker = new Worker(new URL('mgba.js', import.meta.url), {
+        worker = new Worker(globalThis.electron?.isDev ? new URL('mgba.js', import.meta.url) : URL.createObjectURL(new Blob([`await import("${new URL('mgba.js', import.meta.url)}");typeof startWorker !== 'undefined' && startWorker();`], {type: 'text/javascript'})), {
           'type': 'module',
           // This is the way that we signal to the Web Worker that it is hosting
           // a pthread.
